connectors/src/connectors/google_drive/lib.ts

```diff
@@ -7,23 +7,100 @@ import {
   Ok,
 } from "@dust-tt/types";
 import type { InferAttributes, WhereOptions } from "sequelize";
+import { Op } from "sequelize";
 import { v4 as uuidv4 } from "uuid";
 
+import { GOOGLE_DRIVE_WEBHOOK_LIFE_MS } from "@connectors/connectors/google_drive/lib/config";
+import { getGoogleDriveObject } from "@connectors/connectors/google_drive/lib/google_drive_api";
+import {
+  getDrivesIdsToSync,
+  getSyncPageToken,
+} from "@connectors/connectors/google_drive/temporal/activities";
 import { isGoogleDriveSpreadSheetFile } from "@connectors/connectors/google_drive/temporal/mime_types";
-import { getAuthObject } from "@connectors/connectors/google_drive/temporal/utils";
+import {
+  getAuthObject,
+  getDriveClient,
+} from "@connectors/connectors/google_drive/temporal/utils";
 import { HTTPError } from "@connectors/lib/error";
 import {
   GoogleDriveFiles,
   GoogleDriveSheet,
+  GoogleDriveWebhook,
 } from "@connectors/lib/models/google_drive";
+import logger from "@connectors/logger/logger";
 import type { ConnectorResource } from "@connectors/resources/connector_resource";
 import type { ConnectorModel } from "@connectors/resources/storage/models/connector_model";
 
 const { CONNECTORS_PUBLIC_URL, DUST_CONNECTORS_WEBHOOKS_SECRET } = process.env;
 
+export async function registerWebhooksForAllDrives(
+  connector: ConnectorResource,
+  marginMs: number
+): Promise<Result<undefined, Error>> {
+  const driveIdsToSync = await getDrivesIdsToSync(connector.id);
+  const allRes = await Promise.all(
+    driveIdsToSync.map((driveId) => {
+      return ensureWebhookForDriveId(connector, driveId, marginMs);
+    })
+  );
+  for (const res of allRes) {
+    if (res.isErr()) {
+      return res;
+    }
+  }
+
+  return new Ok(undefined);
+}
+
+export async function ensureWebhookForDriveId(
+  connector: ConnectorResource,
+  driveId: string,
+  marginMs: number
+): Promise<Result<string | undefined, Error>> {
+  const webhook = await GoogleDriveWebhook.findOne({
+    where: {
+      driveId: driveId,
+      expiresAt: {
+        [Op.gt]: new Date(new Date().getTime() + marginMs),
+      },
+    },
+  });
+  if (!webhook) {
+    const auth = await getAuthObject(connector.connectionId);
+    const remoteFile = await getGoogleDriveObject(auth, driveId);
+    if (!remoteFile) {
+      throw new Error(`Drive with id ${driveId} not found`);
+    }
+    const res = await registerWebhook(
```

Comment:
Since we've added 2 params to this function, could we have an object as argument? 

```
const res = await registerWebhook({
   connector,
   driveId,
   isInSharedDrive: remoteFile.inSharedDrive
});
```